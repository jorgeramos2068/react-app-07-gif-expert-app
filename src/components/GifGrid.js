import React from 'react';
import PropTypes from 'prop-types';
import GifGridItem from './GifGridItem';
import {useFetchGifs} from '../hooks/useFetchGifs';

const GifGrid = ({category}) => {
  const {data:images, loading} = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p>Loading</p>}
      {<div className="card-grid">
        <ol>
          {
            images.map((image) => (
              <GifGridItem key={image.id} {...image} />
            ))
          }
        </ol>
      </div>}
    </>
  );
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
};

export default GifGrid;