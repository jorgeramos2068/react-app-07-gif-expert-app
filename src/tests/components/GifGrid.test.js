import React from 'react';
import {shallow} from 'enzyme';
import GifGrid from '../../components/GifGrid';
import {useFetchGifs} from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Tests for <GifGrid />', () => {
  let wrapper;
  const category = 'Test';

  test('Should display <GifGrid /> correctly', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should show the list of items when loading via useFetchGifs custom hook', () => {
    const gifs = [{
      id: '1',
      url: 'https://localhost/test.jpg',
      title: 'test'
    }];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    // Chech that <p> does not exist
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
