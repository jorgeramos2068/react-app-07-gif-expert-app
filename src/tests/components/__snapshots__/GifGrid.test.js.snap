// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifGrid /> Should display <GifGrid /> correctly 1`] = `
<Fragment>
  <h3>
    Test
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  >
    <ol />
  </div>
</Fragment>
`;

exports[`Tests for <GifGrid /> Should show the list of items when loading via useFetchGifs custom hook 1`] = `
<Fragment>
  <h3>
    Test
  </h3>
  <div
    className="card-grid"
  >
    <ol>
      <GifGridItem
        id="1"
        key="1"
        title="test"
        url="https://localhost/test.jpg"
      />
    </ol>
  </div>
</Fragment>
`;
